eclair-core/src/main/scala/fr/acinq/eclair/io/MessageRelay.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2021 ACINQ SAS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.io
+
+import akka.actor.{Actor, ActorRef, ActorRefFactory, Props}
```

Comment:
We try to use akka typed for all new actors, it provides stricter types for the messages sent and received (cleaner data contract that can be enforced by the compiler). See https://doc.akka.io/docs/akka/current/typed/index.html for the akka doc, it's always useful to regularly go through it.

A somewhat simple example of how to create a typed actor is the `BlockchainWatchdog.scala`. You have to define a `sealed trait Command` and subtypes for all messages you handle. You'll need to dive in a bit into the concept of message adapter to receive the `PeerConnectionResult` (see https://doc.akka.io/docs/akka/current/typed/interaction-patterns.html#adapted-response).

Reply:
I don't think that's possible without some major changes. The problem is that there is no sender in Akka typed and `Switchboard`, `Peer` and `PeerConnection` all rely on it for the connection.

Reply:
I don't see how this is an issue, we've done that in multiple other actors without any problem?
See for example what we did in the `Channel`:

```
val replyTo = if (c.replyTo == ActorRef.noSender) sender() else c.replyTo
```

Reply:
Done